src/Integration/LocalServices/RuleSetInspector.cs

```diff
@@ -0,0 +1,144 @@
+            //    b. otherwise it will get the Action from the Include
+            //  Once the Include rule Action is determined it will be merged but could only change to more strict value !
+            // 
+            // 3. Lastly the file-level rules are added with their Action (overriding any previous settings).
+            //
+            // To summarize in practical terms: 
+            // (●) IncludeAll - will not impact us because all the baseline rules will be merged on top of (1)
+            //  and will either be more strict or the IncludeAll action will be more strict 
+            //  => either way won't weaken.
+            // 
+            // (●) By default i.e. the way we generate the targetRuleSet (which we expect it to be the project level ruleset),
+            // the includes cannot weaken the base line ruleset (which we expect it to be the solution level ruleset)
+            // due to to the include merge rules which prefer strictness and the fact that we use Default as the Include Action.
+            // 
+            // (●) The only thing that can weaken the rules in practice, in the default case, are ruleset level rules 
+            // i.e what you get when you modify the ruleset by using the ruleset editor UI.
+            // 
+            // (●) In the none default case, when the user decides to manually edit the file, setting the baseline include
+            // to None or removing it will use the other Includes' Action settings which could weaken the target ruleset.
+            // 
+            // With that in mind, we will find "conflicts" of two types - rule with weaker Action or a missing rule.
+            // (assuming that all the rules were found with the help of ruleSetDirectories).
+            // 
+            // In terms of how to fix the conflict...  we could either just add all the conflicted rules into the project. That 
+            // will not be optimal in terms of the user or the rebind/update experience. A better approach would be something like:
+            // 1. Reset the baseline Include to Action=Default
+            // 2. If there are still conflicts, remove all the conflicting rules which are directly on target ruleset.
+            // 3. If there are still conflicts, add the remaining conflicting rules directly under the target with Action=Warning
+
+            var effectiveRulesMap = target.GetEffectiveRules(ruleSetDirectories, providers, this.EffectiveRulesErrorHandler)
+                .ToDictionary(r => r.FullId, StringComparer.OrdinalIgnoreCase);
+
+            var deprioritizedRules = baseLine.Rules.Select(r =>
+            {
+                RuleReference reference;
+                if (effectiveRulesMap.TryGetValue(r.FullId, out reference))
+                {
+                    Debug.Assert(reference.Action != RuleAction.None, "Expected to be found in the missing set");
+
+                    if (reference.Action == RuleAction.Info)
```

Comment:
Yes I did made that assumption. I'll modify the code to support other cases (i.e. was set to warning).